pkg/controller/node/nodecontroller.go

```diff
@@ -238,6 +289,48 @@ func (nc *NodeController) getCondition(status *api.NodeStatus, conditionType api
 	return nil
 }
 
+func (nc *NodeController) checkPod(obj interface{}) {
+	pod, ok := obj.(*api.Pod)
+	if !ok {
+		return
+	}
+	if pod.DeletionTimestamp == nil {
+		return
+	}
+
+	if len(pod.Spec.NodeName) == 0 {
+		nc.forcefullyDeletePod(pod)
+		return
+	}
+
+	nodeObj, found, err := nc.nodeStore.GetByKey(pod.Spec.NodeName)
+	if err != nil {
+		// what do
+		util.HandleError(err)
+		nc.podController.Requeue(pod)
+		return
+	}
+
+	if !found {
+		nc.forcefullyDeletePod(pod)
```

Comment:
I could have sworn we already had code somewhere that deleted pods that are on nodes that no longer exist, but it wasn't in the GC controller so maybe I'm just hallucinating?